### DISCUSSION TITLE
Inspector trait design question

### DISCUSSION THREAD

**Discussion by emnul** [bluealloy/revm#1745] [Q&A] [answered]
I'm curious why EvmContext is not passed to the selfdestruct method within the Inspector trait, but it's passed to all other functions? I noticed even all the way back to when methods in the Inspector took EvmData instead of EvmContext, selfdestruct still only took a target and contract

**Comment by emnul** [comment]
After doing some digging, I realized that the SELFDESTRUCT opcode halts EVM execution and returns the empty tuple `()`. See section 9.4.4 of the [Ethereum Yellow Paper](https://ethereum.github.io/yellowpaper/paper.pdf). With this in mind, my best guess is that `EvmContext` is not passed to the [selfdestruct](https://github.com/bluealloy/revm/blob/fa5650ee8a4d802f4f3557014dd157adfb074460/crates/revm/src/inspector.rs#L165) method because an Inspector should not be able to modify EVM execution via access to EvmContext once this opcode is executed. 

@rakita please let me know if my understand is correct here or if I'm missing anything

**Reply by rakita** [reply, marked as answer]
Hey, the reason for not including Context is rather simple, there were no requests to add it.
If needed you can always use step to filter selfdestruct opcode and act on it

btw there is a Revm tg group. Link is in main page, I would respond there faster
